spatialos-sdk/src/worker/schema.rs

```diff
@@ -413,7 +413,18 @@ impl<T: IndexedField> SchemaField for Option<T> {
     }
 
     fn has_update(update: &ComponentUpdate, field: FieldId) -> bool {
-        T::field_count(update.fields(), field) > 0
+        T::field_count(update.fields(), field) > 0 || update.field_cleared(field)
+    }
+
+    fn get_update(update: &ComponentUpdate, field: FieldId) -> Option<Self::RustType> {
+        if update.field_cleared(field) {
+            Some(None)
+        } else {
+            match Self::get_field(update.fields(), field) {
```

Comment:
Little bit of sugar...

```rust
Self::get_field(update.fields(), field)
	.map(|opt| Some(opt))
```

Reply:
Ah, good catch! I felt like I was missing a better way of writing that 🤦‍♀

Reply:
Also, even better, you can do `.map(Some)`, because `Some` (other tuple-structs) can be used as functions ✨